pkg/controller/cloneset/cloneset_controller.go

```diff
@@ -301,9 +303,23 @@ func (r *ReconcileCloneSet) doReconcile(request reconcile.Request) (res reconcil
 
 	if !isPreDownloadDisabled {
 		if currentRevision.Name != updateRevision.Name {
-			// pre-download images for new revision
-			if err := r.createImagePullJobsForInPlaceUpdate(instance, currentRevision, updateRevision); err != nil {
-				klog.Errorf("Failed to create ImagePullJobs for %s: %v", request, err)
+			// get clone pre-download annotation
+			minUpdatedReadyPodsCount := 0
+			minUpdatedReadyPods, ok := instance.Annotations[appspub.ImagePreDownloadMinUpdatedReadyPods]
+			if ok {
+				if minUpdatedReadyPodsCount, err = strconv.Atoi(minUpdatedReadyPods); err != nil {
```

Comment:
Should we use intOrString to support percentage or just integer?

Reply:
done